frontend/src/pages/onboarding/index.tsx

```diff
@@ -12,117 +12,189 @@ import { ClimbingGradesForm } from './ClimbingGradesForm';
+    form: <FavoriteGymsForm />,
+    validate: {},
+  },
+  {
+    title: 'Your climbing experience',
+    subtitle: 'This can help others to find the right climbing partner',
+    form: <ClimbingGradesForm />,
+    validate: {},
+  },
+  {
+    title: 'Your climbing experience',
+    subtitle: 'This can help others to find the right climbing partner',
+    form: <ClimbingCertForm />,
+    validate: {},
+  },
+  {
+    title: 'Profile Photo',
+    subtitle: 'Upload a profile photo (optional)',
+    form: <AvatarForm />,
+    validate: {},
+  },
+];
+const getFormSchema = (onboardingSteps: OnboardingStep[]): FormSchema =>
+  onboardingSteps.reduce((acc, curr) => ({ ...acc, ...curr.validate }), {});
+const getValidateFields = (activeStep: number): (keyof OnboardingFormValues)[] =>
+  Object.keys(onboardingSteps[activeStep - 1].validate) as (keyof OnboardingFormValues)[];
+
+const formSchema = Yup.object().shape(getFormSchema(onboardingSteps));
+
+const renderTitle = (activeStep: number) => {
+  const { title, subtitle } = onboardingSteps[activeStep - 1];
+
+  return (
+    <>
+      <Typography variant="h4">{title}</Typography>
+      <Typography variant="subtitle1" sx={{ color: 'text.secondary' }}>
+        {subtitle}
+      </Typography>
+    </>
+  );
+};
+const renderForm = (activeStep: number) => onboardingSteps[activeStep - 1].form;
 
 export default function Onboarding() {
   const { enqueueSnackbar } = useSnackbar();
   const navigate = useNavigate();
+  // Ranges from 1-N, where N is the number of steps
   const [activeStep, setActiveStep] = useState<number>(1);
-  const isComplete = activeStep === STEPS.length;
-  const { updateUserIdentity } = useProfile();
+  const isComplete = activeStep === onboardingSteps.length;
 
-  const initialFormValues: UserRequest = useMemo(() => ({}), []);
-  const methods = useForm<UserRequest>({
+  const methods = useForm<OnboardingFormValues>({
     resolver: yupResolver(formSchema),
-    defaultValues: initialFormValues,
-    // Show error onChange, onBlur, onSubmit
     mode: 'onSubmit',
   });
-  const { handleSubmit } = methods;
+  const { handleSubmit, trigger } = methods;
 
-  const _handleSubmit = async (data: UserRequest) => {
-    try {
-      await updateUserIdentity(data);
-      enqueueSnackbar('Successfully completed onboarding!');
+  const { run: submitUploadAvatar } = useSafeRequest(uploadAvatar, {
+    manual: true,
+    onError: (error) => {
+      enqueueSnackbar('Failed to upload profile picture', { variant: 'error' });
+    },
+  });
+  const { run: submitUpdateUser } = useSafeRequest(updateUser, {
+    manual: true,
+    onSuccess: () => {
+      enqueueSnackbar('Successfully completed onboarding.', {
+        autoHideDuration: 5000,
+      });
       navigate(PATH_DASHBOARD.general.jios.root);
+    },
+    onError: (error) => {
+      enqueueSnackbar('Failed to update user', { variant: 'error' });
+    },
+  });
+
+  const _handleSubmitAvatar = async (avatar?: AvatarData) => {
```

Comment:
Wondering the rationale for the _ prefix.

Reply:
will remove the _ from this. 
but _handlesubmit() function has this prefix cos it collides with handlesubmit from useform